internal/driver/gomobile/file.go

```diff
@@ -1,14 +1,90 @@
 package gomobile
 
-import "fyne.io/fyne"
+import (
+	"io"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"fyne.io/fyne"
+)
+
+type fileOpen struct {
+	io.ReadCloser
+	uri  string
+	done func()
+}
+
+//func (f *fileOpen) Save() (io.WriteCloser, error) {
+//	return nativeFileSave(f)
+//}
+
+func (f *fileOpen) Name() string {
+	return nameFromURI(f.uri)
+}
+
+func (f *fileOpen) URI() string {
+	return f.uri
+}
+
+func (d *mobileDriver) FileReaderForURI(uriOrPath string) (fyne.FileReader, error) {
+	uri := uriOrPath
+	if strings.Index(uriOrPath, "://") == -1 {
+		uri = "file://" + uriOrPath
+	}
+
+	file := &fileOpen{uri: uri}
+	read, err := nativeFileOpen(file)
+	file.ReadCloser = read
+	return file, err
+}
+
+type fileSave struct {
+	io.WriteCloser
+	uri string
+}
+
+func (f *fileSave) Name() string {
+	return nameFromURI(f.uri)
+}
+
+func (f *fileSave) URI() string {
+	return f.uri
+}
+
+func (d *mobileDriver) FileWriterForURI(uriOrPath string) (fyne.FileWriter, error) {
+	uri := uriOrPath
+	if strings.Index(uriOrPath, "://") == -1 {
+		uri = "file://" + uriOrPath
+	}
+
+	file := &fileSave{uri: uri}
+	write, err := nativeFileSave(file)
+	file.WriteCloser = write
+	return file, err
+}
```

Comment:
I thought writing is unsupported on mobile?
If so, please add a clarifying comment (or delete).

Reply:
Quite right, I was filling the blanks where really that wasn't necessary. Gone now